src/parser/mod.rs

```diff
@@ -5024,6 +5024,27 @@ impl<'a> Parser<'a> {
                 break;
             }
         }
+
+        // BigQuery accepts any number of quoted identifiers of a table name.
+        // https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#quoted_identifiers
+        if dialect_of!(self is BigQueryDialect)
+            && idents.iter().any(|ident| ident.value.contains('.'))
+        {
+            idents = idents
+                .into_iter()
+                .flat_map(|ident| {
+                    ident
+                        .value
+                        .split('.')
```

Comment:
does BigQuery support any way to escape `.` (like `'foo\.bar'` I wonder 🤔 )

Reply:
BigQuery didn't support escaping dots in identifiers either as far as I could tell and tested